src/NUnitFramework/framework/Attributes/RandomAttribute.cs

```diff
@@ -41,6 +42,25 @@ public class RandomAttribute : DataAttribute, IParameterDataSource
     {
         private RandomDataSource _source;
         private int _count;
+        private bool _distinct;
+
+        /// <summary>
+        /// If true, the random values will be all different
+        /// </summary>
+        public bool Distinct
+        {
+            get
+            {
+                return _distinct;
+            }
+            set
+            {
+                _distinct = value;
+
+                if (_source != null)
+                    _source.Distinct = _distinct;
```

Comment:
Yes, but what I'm saying is [this](https://github.com/jp2masa/nunit/blob/63169cfea2a27a1dad58bc6b97ece92df1c26bfe/src/NUnitFramework/framework/Attributes/RandomAttribute.cs#L214) is the only place I can see the source is used and [this](https://github.com/jp2masa/nunit/blob/63169cfea2a27a1dad58bc6b97ece92df1c26bfe/src/NUnitFramework/framework/Attributes/RandomAttribute.cs#L212) is guaranteed to happen. If you remove lines 60 and 61, I still can't think of any way you could get any data out of the source without causing line 212 to run. What am I missing?